File: backend/app/schemas/draft.py
from __future__ import annotations
import datetime
from typing import Literal, Optional
from pydantic import BaseModel, Field

from app.schemas import DraftStatus


class DraftResponse(BaseModel):
    id: int
    email_id: Optional[int]
    tone: str
    content: str
    recipient: Optional[str] = None
    subject: Optional[str] = None
    status: str
    send_error: Optional[str] = None
    created_at: datetime.datetime
    updated_at: datetime.datetime

    model_config = {"from_attributes": True}


class DraftListResponse(BaseModel):
    items: list[DraftResponse]
    total: int
    page: int
    page_size: int


class DeleteDraftResponse(BaseModel):
    status: str


class DraftPatchRequest(BaseModel):
    content: Optional[str] = None
    recipient: Optional[str] = Field(default=None, min_length=3, max_length=512)
    subject: Optional[str] = Field(default=None, min_length=1, max_length=512)
    status: Optional[DraftStatus] = None


class DraftCreateRequest(BaseModel):
    recipient: str = Field(min_length=3, max_length=512)
    subject: str = Field(min_length=1, max_length=512)
    content: str = Field(min_length=1)


class SendDraftRequest(BaseModel):
    """Optional mailbox selection for sending a reply draft."""

    provider: Optional[Literal["gmail", "outlook"]] = None

File: frontend/src/types/draft.ts
export type DraftTone = 'formal' | 'brief' | 'polite_decline' | 'ask_info' | 'manual';
export type DraftStatus = 'draft' | 'saved' | 'ready_to_send' | 'sent' | 'send_failed' | 'deleted';

export interface DraftResponse {
  id: number;
  email_id: number | null;
  tone: string;
  content: string;
  recipient: string | null;
  subject: string | null;
  status: string;
  send_error: string | null;
  created_at: string;
  updated_at: string;
}

export interface DraftListResponse {
  items: DraftResponse[];
  total: number;
  page: number;
  page_size: number;
}

export interface DraftPatchRequest {
  content?: string;
  recipient?: string;
  subject?: string;
  status?: DraftStatus;
}

export interface DraftCreateRequest {
  recipient: string;
  subject: string;
  content: string;
}
